src/queries.jl

```diff
@@ -2,28 +2,28 @@
 
 Assign a tag to a slot in a register.
 
-It returns the list of all currently present tags for that register.
-
 See also: [`query`](@ref), [`untag!`](@ref)"""
-function tag!(ref::RegRef, tag::Tag)
-    push!(ref.reg.tags[ref.idx], tag)
+function tag!(ref::RegRef, tag::Tag; tag_time::Union{Float64, Nothing}=nothing, id::Union{Int128, Nothing}=nothing)
```

Comment:
- is `tag_time` ever used by any code we currently have?
- is `id` ever used by any code we currently have?

If not, let's remove the options to have them as input arguments and simplify the body of this function

Reply:
nope, they're only there just to provide the option to the user to override the guid function, but since this is an internal implementation detail, it would make sense to remove these